pkg/dryrun/dryrun_utils.go

```diff
@@ -0,0 +1,264 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package dryrun
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"sort"
+
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+
+	policyv1 "open-cluster-management.io/config-policy-controller/api/v1"
+)
+
+var (
+	successColor = color.New(color.FgGreen).SprintFunc()
+	errorColor   = color.New(color.FgRed).SprintFunc()
+	warningColor = color.New(color.FgYellow).SprintFunc()
+)
+
+func compareStatus(cmd *cobra.Command,
+	inputStatus map[string]interface{},
+	resultStatus map[string]interface{},
+) {
+	cmd.Println("# Status compare:")
+
+	isStatusMatch := compareStatusObj(cmd, true, "", inputStatus, resultStatus)
+
+	if isStatusMatch {
+		cmd.Println(successColor("\033[1m Expected status matches the actual status \033[0m"))
+	} else {
+		cmd.Println(errorColor("\033[1m Expected status does not match the actual status \033[0m"))
+	}
+}
+
+// compareStatusObj compares the expected inputMap against the actual resultMap
+// and prints success or error messages for mismatches. It supports nested structures,
+// including maps and arrays, by recursively comparing their contents. The function
+// updates isStatusMatch based on whether all expected values match the actual values.
+func compareStatusObj(cmd *cobra.Command, isStatusMatch bool, parentPath string,
+	inputMap, resultMap map[string]interface{},
+) bool {
+	// Sort keys for consistency
+	keys := make([]string, 0, len(inputMap))
+	for k := range inputMap {
+		keys = append(keys, k)
+	}
+
+	sort.Strings(keys)
+
+	for _, key := range keys {
+		actualValue, exist := resultMap[key]
+		value := inputMap[key]
+
+		if !exist {
+			cmd.Println(errorColor(fmt.Sprintf("Key '%s' not found in result at path '%s'", key, parentPath)))
+
+			isStatusMatch = false
+
+			continue
+		}
+
+		path := parentPath + "." + key
+
+		switch typedV := value.(type) {
+		case string, int, int64, float64, bool:
+			v := fmt.Sprintf("%v", typedV)
+
+			av := fmt.Sprintf("%v", actualValue)
+
+			if v != av {
+				errorMatchPrint(cmd, path, v, av)
+
+				isStatusMatch = false
+			} else {
+				successMatchPrint(cmd, path, v, av)
+			}
+
+		case []interface{}:
```

Comment:
For readability, is it at all possible to split this case into another function? Or maybe just split the map case in the switch to another function?